In [1]:
# loading the libraries
import pandas as pd
from scipy.stats import chi2_contingency, ttest_ind, mannwhitneyu

In [2]:
# Loading data
df = pd.read_csv(r"data\no_show_cleaned.csv")

## TEST 1: Waiting Days vs No-Show

#### Hypothesis
* H₀ (null): Waiting time has no effect on no-show

* H₁ (alt): Waiting time affects no-show

In [3]:
show = df[df['no_show'] == 0]['waiting_days']
no_show = df[df['no_show'] == 1]['waiting_days']

stat, p_value = mannwhitneyu(show, no_show)

print("P-value:", p_value)

P-value: 4.975209927689346e-122


#### Interpretation
There is a statistically significant relationship between waiting days and no-show (p < 0.05).

#### Conclusion
Reject H₀. Waiting time significantly affects whether a patient will no-show.

## TEST 2: Age vs No-Show
#### Hypothesis
* H₀: Age has no effect
* H₁: Age affects attendance

In [4]:
show_age = df[df['no_show'] == 0]['age']
no_show_age = df[df['no_show'] == 1]['age']

stat, p_value = mannwhitneyu(show_age, no_show_age)

print("P-value:", p_value)

P-value: 2.7490453030239144e-171


#### Interpretation
There is a statistically significant relationship between age and no-show (p < 0.05).

#### Conclusion
Reject H₀. Age significantly impacts appointment attendance.

## TEST 3: SMS vs No-Show 
###### (Categorical variable - Chi-Square)
#### Hypothesis
* H₀: SMS has no relationship with no-show
* H₁: SMS is related to no-show

In [5]:
contingency = pd.crosstab(df['sms_received'], df['no_show'])

chi2, p_value, _, _ = chi2_contingency(contingency)

print("P-value:", p_value)

P-value: 3.275027505140042e-08


#### Interpretation
There is a statistically significant relationship between SMS reminders and no-show (p < 0.05).

#### Conclusion
Reject H₀. SMS reminders are significantly related to no-show behavior.

## TEST 4: Gender vs No-Show
#### Hypothesis
* H₀: Gender has no relationship with no-show
* H₁: Gender is related to no-show

In [6]:
contingency = pd.crosstab(df['gender'], df['no_show'])

chi2, p_value, _, _ = chi2_contingency(contingency)

print("P-value:", p_value)

P-value: 0.5441949418015478


#### Interpretation
There is no statistically significant relationship between gender and no-show (p > 0.05).

#### Conclusion
Fail to reject H₀. Gender does not predict whether a patient will miss an appointment.

## TEST 5: Scholarship vs No-Show
#### Hypothesis
* H₀: Scholarship has no relationship with no-show
* H₁: Scholarship is related to no-show

In [7]:
contingency = pd.crosstab(df['scholarship'], df['no_show'])

chi2, p_value, _, _ = chi2_contingency(contingency)

print("P-value:", p_value)

P-value: 1.866839149761171e-34


#### Interpretation
There is a statistically significant relationship between scholarship status and no-show (p < 0.05).

#### Conclusion
Reject H₀. Scholarship status significantly relates to no-show behavior.

## TEST 6: Medical Conditions
#### Hypothesis
* H₀: Medical conditions (hypertension, diabetes, alcoholism) have no relationship with no-show
* H₁: Medical conditions are related to no-show

In [8]:
for col in ['hypertension', 'diabetes', 'alcoholism']:
    contingency = pd.crosstab(df[col], df['no_show'])
    chi2, p_value, _, _ = chi2_contingency(contingency)
    print(col, "p-value:", p_value)

hypertension p-value: 1.8519489386187532e-52
diabetes p-value: 2.0172863962868155e-09
alcoholism p-value: 1.1444400585517945e-07


#### Interpretation
There is a statistically significant relationship between medical conditions (hypertension, diabetes, alcoholism) and no-show (p < 0.05).

#### Conclusion
Reject H₀. Medical conditions are significantly associated with no-show behavior.

## TEST 7: Weekday Effect
#### Hypothesis
* H₀: Weekday has no relationship with no-show
* H₁: Weekday is related to no-show

In [9]:
contingency = pd.crosstab(df['appointment_weekday'], df['no_show'])

chi2, p_value, _, _ = chi2_contingency(contingency)

print("P-value:", p_value)

P-value: 1.868506115060262e-09


#### Interpretation
There is a statistically significant relationship between weekday and no-show (p < 0.05).

#### Conclusion
Reject H₀. The day of the week significantly affects no-show rates.

## FINAL SUMMARY

This analysis examined 7 factors to determine their relationship with patient no-show behavior:

| Test | Factor | P-value | Significant? |
|------|--------|---------|---------------|
| 1 | Waiting Days | 4.98e-122 | Yes |
| 2 | Age | 2.75e-171 | Yes |
| 3 | SMS Reminders | 3.28e-08 | Yes |
| 4 | Gender | 0.544 | No |
| 5 | Scholarship | 1.87e-34 | Yes |
| 6 | Medical Conditions | < 0.001 | Yes |
| 7 | Weekday | 1.87e-09 | Yes |

**Key Findings:**

1. **6 out of 7 factors showed statistically significant relationships** with no-show behavior.
2. **Gender is the only non-significant predictor** - male and female patients have similar no-show rates.

**Recommendations:**
- Focus on SMS reminder optimization to reduce no-shows
- Consider reducing waiting time between booking and appointment
- Develop targeted interventions for specific age groups and weekdays
- Monitor patients with chronic conditions (hypertension, diabetes, alcoholism)
- Scholarship holders have different attendance patterns - investigate reasons